docs/source/format/CanonicalExtensions.rst

```diff
@@ -283,6 +283,132 @@ UUID
    A specific UUID version is not required or guaranteed. This extension represents
    UUIDs as FixedSizeBinary(16) with big-endian notation and does not interpret the bytes in any way.
 
+Unknown
+=======
+
+Unknown represents a type or array that an Arrow-based system received from an
+external (often non-Arrow) system, which it cannot interpret itself or did not
+have support for in advance.  In this case, it can pass on Unknown to its own
+clients to communicate that a field exists, but that it cannot interpret the
+field or data.
+
+Extension parameters:
+
+* Extension name: ``arrow.unknown``.
```

Comment:
On name bike shedding  I'll add `arrow.unsupported` or `arrow.external` given it is coming from an external system.  Unsupported, at least to me, reinforces the point that there shouldn't a convention around this.